agent/exporter/src/main/java/com/microsoft/applicationinsights/agent/Exporter.java

```diff
@@ -412,34 +402,41 @@ private static void setProperties(Map<String, String> properties, long timeEpoch
         }
     }
 
-    private static void applyHttpRequestSpan(Map<AttributeKey<?>, Object> attributes, RemoteDependencyTelemetry telemetry) {
-
-        telemetry.setType("Http (tracked component)");
-
-        String method = removeAttributeString(attributes, SemanticAttributes.HTTP_METHOD);
-        String url = removeAttributeString(attributes, SemanticAttributes.HTTP_URL);
+    private static void applyHttpRequestSpan(Map<AttributeKey<?>, Object> attributes, RemoteDependencyTelemetry telemetry, String httpMethod) {
 
         Object httpStatusCode = attributes.remove(SemanticAttributes.HTTP_STATUS_CODE);
         if (httpStatusCode instanceof Long) {
             telemetry.setResultCode(Long.toString((Long) httpStatusCode));
         }
 
+        String target = removeAttributeString(attributes, SemanticAttributes.PEER_SERVICE);
+        if (target == null) {
+            target = removeAttributeString(attributes, SemanticAttributes.NET_PEER_NAME);
+        }
+        if (target == null) {
+            target = removeAttributeString(attributes, SemanticAttributes.NET_PEER_IP);
+        }
+        String targetAppId = removeAttributeString(attributes, SPAN_TARGET_ATTRIBUTE_NAME);
+        if (targetAppId == null || AiAppId.getAppId().equals(targetAppId)) {
+            telemetry.setType("HTTP");
+            telemetry.setTarget(target);
+        } else {
+            // using "Http (tracked component)" is important for dependencies that go cross-component (have an appId in their target field)
+            // if you use just HTTP, Breeze will remove appid from the target
+            telemetry.setType("Http (tracked component)");
+            telemetry.setTarget(target + " | " + targetAppId);
+        }
+
+        String url = removeAttributeString(attributes, SemanticAttributes.HTTP_URL);
         if (url != null) {
             try {
                 URI uriObject = new URI(url);
-                String target = createTarget(uriObject);
```

Comment:
createTarget appends port to be part of the target.. i don't see that get added back.. are we dropping port entirely?

Reply:
good catch! yes, we want to keep the port, i will fix 👍